scripts/generate_pkcs7_tests.py

```diff
@@ -37,7 +39,15 @@ class TestData:
     Take in test_suite_pkcs7.data file.
     Allow for new tests to be added.
     """
-    mandatory_dep = "MBEDTLS_MD_CAN_SHA256"
+    # In Mbed TLS 3.6, PKCS7 code belongs to the legacy domain
+    # (see transition-guards.md for more information) thus use
+    # MBEDTLS_MD_CAN_SHA256 dependency symbol instead of PSA_WANT_ALG_SHA_256.
+    mandatory_dep = "MBEDTLS_MD_CAN_SHA256" if is_mbedtls_3_6() else "PSA_WANT_ALG_SHA_256"
+
+    # OR helper function but it does not help with the difficulty to decide
+    # if some legacy domain code is involved or not.
```

Comment:
The caller knows what domain it's in. For example PKCS7 is in the USE_PSA_CRYPTO domain.

Reply:
That's the difficult part to me actually. I thought it was the legacy domain.

Reply:
All of X.509, TLS and PK are in the USE domain. [USE and legacy are equivalent for hashes though](https://github.com/Mbed-TLS/mbedtls/blob/a0465779c9b3d11304549ba94f3bf92fca12eec0/docs/architecture/psa-migration/transition-guards.md#hashes): both use `MBEDTLS_MD_CAN_xxx`.

Reply:
> That's the difficult part to me actually.

I think that's the difficult part for everyone :) I was involved in creating these macros and I still need to check from time to time!

> All of X.509, TLS and PK are in the USE domain.

I think some parts or TLS 1.3 (those that are not shared with 1.2) are not in the USE domain, but instead always use PSA. But for PK and X.509 yes.